src/scms/git.js

```diff
@@ -73,15 +73,34 @@ export const getChangedFiles = (directory, revision, staged) => {
     ...(staged
       ? []
       : getLines(
-          runGit(directory, ['ls-files', '--others', '--exclude-standard']),
+          await runGit(directory, [
+            'ls-files',
+            '--others',
+            '--exclude-standard',
+          ]),
         )),
   ].filter(Boolean);
 };
 
-export const getUnstagedChangedFiles = (directory) => {
-  return getChangedFiles(directory, null, false);
+export const getUnstagedChangedFiles = async (directory) => {
+  return await getChangedFiles(directory, null, false);
 };
 
-export const stageFile = (directory, file) => {
-  runGit(directory, ['add', file]);
+export const stageFiles = async (directory, files) => {
+  const maxArguments = 100;
```

Comment:
Complete transparency: I didn't spend too much time looking into `exaca`, but I know that some OS/shells limit the number of arguments you can pass. I figured 100 is a reasonable batch size